kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/util/Metrics.java

```diff
@@ -288,4 +293,14 @@ public static void clear() {
         CLIENT_TO_PROXY_CONNECTION_CACHE.clear();
         PROXY_TO_SERVER_CONNECTION_CACHE.clear();
     }
+
+    public static void bindNettyEventExecutorMetrics(final EventLoopGroup eventLoopGroup) {
+        new NettyEventExecutorMetrics(eventLoopGroup).bindTo(io.micrometer.core.instrument.Metrics.globalRegistry);
+    }
```

Comment:
The underlying binder supports passing `Iterable<EventExecutor>` so we could do 
```suggestion
    public static void bindNettyEventExecutorMetrics(final EventLoopGroup... eventLoopGroup) {
        new NettyEventExecutorMetrics(Arrays.stream(eventLoopGroup).map(EventExecutor.class::cast).collect(Collectors.toList())).bindTo(globalRegistry);
    }
```

Which makes me wonder if `EventExecutor` is a better interface for the `EventGroupConfig` to depend on rather  than `EventLoopGroup` but thats a much bigger refactoring than we need to think about here.

Reply:
Regardless of the above suggestion we already have a static import for `globalRegistry` so we don't need the fully qualified name here.

Reply:
The inheritance hierarchy of `EventExecutorGroup` and `EventExecutor` is kind of cyclic.

`EventExecutorGroup` extends `Iterable<EventExecutor>` and `EventExecutor` extends `EventExecutorGroup` so when I try to cast as suggested I end up getting a `ClassCastException`. 

```log
java.lang.ClassCastException: Cannot cast io.netty.channel.kqueue.KQueueEventLoopGroup to io.netty.util.concurrent.EventExecutor
```

But I get the idea of your suggestion and see how I can implement it.